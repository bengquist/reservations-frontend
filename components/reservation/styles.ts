import styled, { css } from "styled-components";

export const Form = styled.form`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 2rem;

  @media (max-width: 500px) {
    grid-template-columns: auto;
  }
`;

export const ButtonContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 0.5rem;
  justify-self: end;
  align-self: end;
`;

export const secondaryStyles = css`
  background: white;
  color: black;

  :focus {
    color: white;
  }
`;

export const Message = styled.label`
  background: ${props => props.theme.colors.accent};
  grid-column: 1/3;
  grid-row: 3;
  color: white;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0 auto;
`;
