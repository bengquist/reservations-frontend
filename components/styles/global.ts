import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  html {
    font-size: 10px;
  }
  body {
    font-size: 1.6rem;
  }
  :root {
    --step-up-5: 2em;
    --step-up-4: 1.7511em;
    --step-up-3: 1.5157em;
    --step-up-2: 1.3195em;
    --step-up-1: 1.1487em;
    /* baseline: 1em */
    --step-down-1: 0.8706em;
    --step-down-2: 0.7579em;
    --step-down-3: 0.6599em;
    --step-down-4: 0.5745em;
    --step-down-5: 0.5em;
    /* Colors */
    --header: rgb(0,0,0);
  }
  @font-face {
    font-family: system;
    font-style: normal;
    font-weight: 300;
    src: local(".SFNSText-Light"), local(".HelveticaNeueDeskInterface-Light"), local(".LucidaGrandeUI"), local("Ubuntu Light"), local("Segoe UI Light"), local("Roboto-Light"), local("DroidSans"), local("Tahoma");
  }
  body, h1, h2, h3, h4, h5, h6, p, ol, ul, input[type=text], input[type=email], button {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }
  body, h1, h2, h3, h4, h5, h6, p, ol, ul, input[type=text], input[type=email], button {
    font-family: "system"
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ol, ul {
    list-style: none;
  }
  img {
    max-width: 100%;
    height: auto;
  }
  /* Links */
  a {
    text-decoration: none;
    color: inherit;
  &.active {
      text-decoration: none;
    }
  }
`;
