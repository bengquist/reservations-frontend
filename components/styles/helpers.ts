import { css } from "styled-components";

export const hoverState = css`
  transition: 0.2s;
  :hover,
  :active {
    background-color: ${({ theme }) => theme.colors.primary};
  }
`;
