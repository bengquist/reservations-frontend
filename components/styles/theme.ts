export type AppTheme = {
  colors: {
    primary: string;
    secondary: string;
    accent: string;
    gray: string;
  };
};

export type AppThemeColor = keyof AppTheme["colors"];

export const theme: AppTheme = {
  colors: {
    primary: "#DAF2FC",
    secondary: "#009CDE",
    accent: "#104C97",
    gray: "#E8E8E8"
  }
};
