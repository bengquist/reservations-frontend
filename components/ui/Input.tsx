import { css } from "styled-components";

function Input(props) {
  return <input css={inputStyle} {...props} />;
}

export default Input;

export const inputStyle = css`
  border-radius: 5px;
  transition: 0.2s;
  padding: 0.5rem;

  :focus {
    border: none;
    box-shadow: 0 0 0 1pt
      ${props => props.theme.colors && props.theme.colors.secondary};
  }
`;
