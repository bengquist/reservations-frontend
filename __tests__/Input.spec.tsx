import * as React from "react";
import { mount } from "enzyme";
import { ThemeProvider } from "styled-components";
import { theme } from "../components/styles/theme";
import Input from "../components/ui/Input";

const setup = (props?: React.ComponentPropsWithoutRef<"input">) =>
  mount(
    <ThemeProvider theme={theme}>
      <Input {...props} />
    </ThemeProvider>
  );

describe("Input", () => {
  it("renders and matches snapshot", () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("contains input element", () => {
    const wrapper = setup();

    const input = wrapper.find("input");

    expect(input).not.toBeNull();
  });

  it("accepts onChange prop", () => {
    const wrapper = setup({ onChange: jest.fn });

    const input = wrapper.find("input");

    expect(input.prop("onChange")).toEqual(jest.fn);
  });

  it("toggles onChange prop", () => {
    const onChange = jest.fn();

    const wrapper = setup({ onChange, value: "" });

    const input = wrapper.find("input");

    input.simulate("change", { target: { value: "Hilton" } });

    expect(onChange).toHaveBeenCalled();
  });
});
