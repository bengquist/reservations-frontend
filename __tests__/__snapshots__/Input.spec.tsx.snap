// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders and matches snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "accent": "#104C97",
        "gray": "#E8E8E8",
        "primary": "#DAF2FC",
        "secondary": "#009CDE",
        "white": "#fff",
      },
      "shadows": Object {
        "soft": "0px 3px 6px rgba(0, 0, 0, 0.15)",
      },
    }
  }
>
  <Input>
    <input
      css={
        Array [
          "
  border-radius: 5px;
  transition: 0.2s;
  padding: 0.5rem;

  :focus {
    border: none;
    box-shadow: 0 0 0 1pt ",
          [Function],
          ";
  }
",
        ]
      }
    />
  </Input>
</ThemeProvider>
`;
